File: app/src/main/java/com/denchic45/kts/uivalidator/util/PhoneNumValid.java
package com.denchic45.kts.uivalidator.util;

import com.denchic45.kts.utils.Validations;

import java.util.function.Supplier;

public class PhoneNumValid implements Supplier<Boolean> {

    private final String phoneNum;

    public PhoneNumValid(String phoneNum) {
        this.phoneNum = phoneNum;
    }

    @Override
    public Boolean get() {
        return Validations.isValidPhoneNumber(phoneNum);
    }
}

File: app/src/main/java/com/denchic45/kts/uivalidator/util/PhoneNumValid.kt
package com.denchic45.kts.uivalidator.util

import com.denchic45.kts.utils.Validations.validPhoneNumber
import com.denchic45.kts.utils.Validations
import java.util.function.Supplier

class PhoneNumValid(private val phoneNum: String) : Supplier<Boolean> {
    override fun get(): Boolean {
        return validPhoneNumber(phoneNum)
    }
}